src/popup/Popup.tsx

```diff
@@ -0,0 +1,202 @@
+                <Text span inherit c="var(--mantine-color-anchor)">
+                    links to stuff
+                </Text>{" "}
+                and some settings. Right now we only have one.
+            </Text>
+            <EnabledSwitch />
+        </>
+    );
+}
+
+function EnabledSwitch() {
+    const [checked, setChecked] = React.useState(false);
+    const fetchIsEnabled = React.useCallback(async () => {
+        try {
+            const isEnabled = await isExtensionEnabled();
+            setChecked(isEnabled ?? false);
+        } catch (e) {
+            console.error("Could not read `isEnabled`", e);
+        }
+    }, []);
+
+    React.useEffect(() => { fetchIsEnabled(); }, [fetchIsEnabled]);
+
+    const toggleEnabled = React.useCallback(async () => {
+        await setIsExtensionEnabled(!checked);
+        fetchIsEnabled();
+    }, [checked, fetchIsEnabled]);
+
+    return (
+        <Switch
+            checked={checked}
+            onChange={toggleEnabled}
+            label={"Enabled"}
+        />
+    );
+}
+
+const SearchPage: React.FC<{}> = () => {
+    const [companyNames, setcompanyNames] = React.useState<"loading" | "error" | string[]>("loading");
+    const fetchCompanies = React.useCallback(async () => {
+        try {
+            const companyData = await getCompanyData();
+            setcompanyNames(companyData.companies);
+        } catch (e) {
+            console.error("Could not read `company data`", e);
+            // setcompanyNames([]);
+            setcompanyNames("error");
+        }
+    }, []);
+
+    React.useEffect(() => { fetchCompanies(); }, [fetchCompanies]);
+
+    if (companyNames == "loading") {
+        return <Loader />
+    }
+
+    // TODO: maybe expose means to handle errors here
+    if (companyNames == "error") {
+        return (
+            <div className="padded">
+                <Notification withCloseButton={false} color="red" title="Error">
+                    Failed to read data from chrome's local storage.
+                </Notification>
+            </div>
+        );
+    }
+
+    return companyNames.length > 0 ? (<CompaniesPanel companyNames={companyNames} />) : (<DownloadCSVPanel setcompanyNames={setcompanyNames} />);
+};
+
+const CompaniesPanel: React.FC<{ companyNames: string[] }> = ({ companyNames }) => {
+    const [searchText, setSearchText] = React.useState("");
+    const filteredCompanyNames = React.useMemo(() => {
+        console.log('searchText', searchText);
+        console.log('companyNames.lengths', companyNames.length);
+        const lowerCaseText = searchText.toLowerCase();
+        if (searchText === "") {
+            return companyNames;
+        }
+        return companyNames.filter(name => name.toLowerCase().includes(lowerCaseText));
+    }, [companyNames, searchText]);
+
+    const tableData: TableData = React.useMemo(() => ({
+        body: filteredCompanyNames.slice(0, 50).map(name => [<Text key={name}>{name}</Text>]),
```

Comment:
`<Text .../>` here should probably use smaller font size. The table rows are almost as big as the panel's title o.O

<img width="402" alt="image" src="https://github.com/irmk323/sponsor-checker/assets/32279273/3fdfb5ca-c364-4c51-81f3-afd11954c9a5">